Automattic/wp-calypso
# Emails: enable the button to add emails so we rely on messaging in em…

#### Proposed Changes

This change will have the "Add Email" button always active. If there is some reason why the user is not able to add an email, then we will rely on the information displayed in the Email Comparison screen.

I've also taken the opportunity to convert the React Class Component into a functional react in TypeScript.

In order to ease the review, the main change was to remove a flag that was disabling / not rendering the button to add email, the referenced flag is `canCurrentUserAddEmail( domain )`
```
{ canCurrentUserAddEmail( domain ) && (
       <Button
 	      href={ emailManagementPurchaseNewEmailAccount(
 		      selectedSiteSlug,
 		      domain.name,
 		      currentRoute,
 		      source
 	      ) }
       >
 	      { translate( 'Add Email' ) }
       </Button>
) }
```

So newly created function component in TypeScript, will no longer have the ` canCurrentUserAddEmail( domain )` check

#### Testing Instructions

1. Have a domain that does not belong to you in your site (add a user to your site, and transfer the domain within calypso)
2. Assert that you can add email for that domain and you see the message about why you can't add an email to the domain.

*

#### Pre-merge Checklist






Related to #
